snuba/query/processors/mapping_promoter.py

```diff
@@ -107,7 +112,7 @@ def transform_nested_column(exp: Expression) -> Expression:
                     # function when the promoted column is not a string since the
                     # supported values of mapping columns are strings and the clients
                     # expect such.
-                    if (
+                    if not self.__cast_to_string or (
```

Comment:
I feel like the logic here is reversed. If `cast_to_string` is True, I would expect the code to cast it to a String, instead of skipping.

Reply:
Isn't that what's happening? This is the branch that doesn't do the cast and it is executed if cast_to_string if false.

Reply:
Oh my mistake. Comment withdrawn.